pkg/server/structlogging/hot_ranges_log_test.go

```diff
@@ -101,72 +90,86 @@ func TestHotRangesStats(t *testing.T) {
-
-		testutils.SucceedsSoon(t, func() error {
-			logs := spy.Logs()
-			if len(logs) < 5 {
-				return errors.New("waiting for hot ranges to be logged")
-			}
+	ts := tc.ApplicationLayer(0)
+	return ts, spy, teardown
+}
 
-			return nil
-		})
-		structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, 1*time.Hour)
+// TestHotRangesStatsTenants tests that hot ranges stats are logged per node.
+// The test will ensure each node contains 5 distinct range replicas for hot
+// ranges logging. Each node should thus log 5 distinct range ids.
+func TestHotRangesStats(t *testing.T) {
+	skip.UnderRace(t)
+	ctx := context.Background()
+	ts, spy, teardown := setupHotRangesLogTest(t, ctx)
+	defer teardown()
+
+	structlogging.TelemetryHotRangesStatsEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
+	structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, time.Millisecond)
+	structlogging.TelemetryHotRangesStatsLoggingDelay.Override(ctx, &ts.ClusterSettings().SV, 0*time.Millisecond)
+
+	structlogging.HotRangeLogManualTicker <- struct{}{}
+	testutils.SucceedsSoon(t, func() error {
+		logs := spy.Logs()
 
-		// Get first 5 logs since the logging loop may have fired multiple times.
-		// We should have gotten 5 distinct range ids, one for each split point above.
-		logs := spy.Logs()[:5]
+		// Depend on a range which we don't exist to go anywhere.
 		for _, l := range logs {
-			assert.Equal(t, l.Databases, []string{"‹test›"})
-			assert.Equal(t, l.Tables, []string{"‹foo›"})
-			assert.Equal(t, l.Indexes, []string{"‹foo_pkey›"})
-			_, ok := rangeIDs[l.RangeID]
-			if ok {
-				t.Fatalf(`Logged ranges should be unique per node for this test.
-found range on node %d and node %d: %s %s %s %s %d`, i, l.LeaseholderNodeID, l.Databases, l.SchemaName, l.Tables, l.Indexes, l.RangeID)
+			if !slices.Equal(l.Databases, []string{"‹system›"}) {
+				continue
 			}
-			rangeIDs[l.RangeID] = struct{}{}
+			if !slices.Equal(l.Tables, []string{"‹sqlliveness›"}) {
+				continue
+			}
+			if !slices.Equal(l.Indexes, []string{"‹primary›"}) {
+				continue
+			}
+			return nil
 		}
+		return errors.New("waited too long for the synthetic data")
+	})
+}
 
+func TestHotRangeLogIntervalSetting(t *testing.T) {
+	skip.UnderRace(t)
+	ctx := context.Background()
+	ts, spy, teardown := setupHotRangesLogTest(t, ctx)
+	defer teardown()
+
+	intervalDuration := 50 * time.Millisecond
+	start := time.Now()
+	structlogging.TelemetryHotRangesStatsEnabled.Override(ctx, &ts.ClusterSettings().SV, true)
+	structlogging.TelemetryHotRangesStatsInterval.Override(ctx, &ts.ClusterSettings().SV, intervalDuration)
+	structlogging.TelemetryHotRangesStatsLoggingDelay.Override(ctx, &ts.ClusterSettings().SV, 0*time.Millisecond)
+
+	// very that there's no logged hot ranges, despite the system ticking
+	structlogging.HotRangeLogManualTicker <- struct{}{}
+	if time.Since(start) > intervalDuration {
```

Comment:
This part seems a little fragile. From what I understand we want to test that no logs are sent before we've waited out the duration. Maybe there's a better way to do this. Can we check timestamps in the log spy? 

Although, if we move to mock the ticker directly I think that might be more straightforard. I'm not really sure what the purpose of the manual ticker is.

Reply:
This test we can actually get rid of. The ranges themselves actually won't come online before some period of time - so any waits may not do what we want.